tracer/src/Datadog.Trace/Ci/GitCommandHelper.cs

```diff
@@ -188,6 +189,222 @@ static List<FileCoverageInfo> ParseGitDiff(string diffOutput)
+
+                if (branchOutput is not { ExitCode: 0 } || string.IsNullOrWhiteSpace(branchOutput?.Output))
+                {
+                    Log.Warning("GitCommandHelper: Failed to get current branch");
+                    return null;
+                }
+
+                targetBranch = branchOutput!.Output.Trim();
+            }
+
+            // Verify branch exists
+            var verifyBranchOutput = RunGitCommand(
+                workingDirectory,
+                $"rev-parse --verify --quiet {targetBranch}",
+                MetricTags.CIVisibilityCommands.VerifyBranchExists);
+
+            if (verifyBranchOutput is not { ExitCode: 0 })
+            {
+                Log.Warning("GitCommandHelper: Branch '{Branch}' does not exist", targetBranch);
+                return null;
+            }
+
+            // 2. Check if the target is already a main-like branch
+            string shortTargetName = targetBranch!;
+            if (shortTargetName.StartsWith($"{remoteName}/"))
+            {
+                shortTargetName = shortTargetName.Substring(remoteName.Length + 1);
+            }
+
+            if (Regex.IsMatch(shortTargetName, branchFilterPattern))
+            {
+                Log.Debug("GitCommandHelper: Branch '{Branch}' already matches branch filter → no parent needed", targetBranch);
+                return null;
+            }
+
+            // 3. Detect default branch
+            string? defaultBranch = null;
+
+            // Try to get the symbolic reference for origin/HEAD
+            var symbolicRefOutput = RunGitCommand(
+                workingDirectory,
+                $"symbolic-ref --quiet --short refs/remotes/{remoteName}/HEAD",
+                MetricTags.CIVisibilityCommands.GetSymbolicRef);
+
+            if (symbolicRefOutput is { ExitCode: 0 } && !string.IsNullOrWhiteSpace(symbolicRefOutput.Output))
+            {
+                var symbolicRef = symbolicRefOutput.Output.Trim();
+                string prefix = $"{remoteName}/";
+                if (symbolicRef.StartsWith(prefix))
+                {
+                    defaultBranch = symbolicRef.Substring(prefix.Length);
+                }
+                else
+                {
+                    defaultBranch = symbolicRef;
+                }
+            }
+            else
+            {
+                // Fallback to main or master
+                foreach (var fallback in new[] { "main", "master" })
+                {
+                    var fallbackOutput = RunGitCommand(
+                        workingDirectory,
+                        $"show-ref --verify --quiet refs/remotes/{remoteName}/{fallback}",
+                        MetricTags.CIVisibilityCommands.ShowRef);
+
+                    if (fallbackOutput is { ExitCode: 0 })
+                    {
+                        defaultBranch = fallback;
+                        break;
+                    }
+                }
+            }
```

Comment:
We want to get the default branch from the backend right?

Reply:
You're right! I completely forgot it, I'll add the default branch in the settings response and delete this